src/node_url.cc

```diff
@@ -407,29 +407,66 @@ int64_t ParseIPv4Number(const char* start, const char* end) {
   return strtoll(start, nullptr, R);
 }
 
+// https://url.spec.whatwg.org/#ipv4-number-parser
+bool IsIPv4NumberValid(std::string_view input) {
+  if (input.empty()) {
```

Comment:
This is unnecessary. This is already handled on the caller function. Better to add a debug assert in here for the input.empty case